python/cudf_polars/cudf_polars/dsl/ir.py

```diff
@@ -690,7 +714,12 @@ def slice_skip(tbl: plc.Table) -> plc.Table:
                 null_order=plc.types.NullOrder.AFTER,
                 name=name,
             )
-            df = DataFrame([index_col, *df.columns])
+            if next(iter(schema)) != name:
+                name_col_map = {col.name: col for col in [index_col, *df.columns]}
+                columns = [name_col_map[name] for name in schema]
+            else:
+                columns = [index_col, *df.columns]
+            df = DataFrame(columns)
```

Comment:
Can you show an example that would fail and is fixed by this? AIUI, the selection that reorders the position of the index appears as a separate node in the IR we receive from polars.

Reply:
Ah, not for parquet reads, I see.

I think we should write this like:

```
df = DataFrame([index_col, *df.columns])
if next(iter(schema)) != name:
    df = df.select(schema)
```

Reply:
Done in https://github.com/rapidsai/cudf/pull/18581/commits/2b76c10e3fc8b8c4eb0a0f8bcb12580a384ceaab